catalog/dags/data_refresh/remap_table_indices.py

```diff
@@ -0,0 +1,137 @@
+"""
+# Remap Table Indices
+
+This file contains TaskGroups related to rmampping the table indices from the live media table onto the temp table to prepare it for promotion.
+"""
+
+import logging
+from typing import NamedTuple
+
+from airflow.decorators import task, task_group
+from airflow.utils.trigger_rule import TriggerRule
+
+from common.sql import fetch_all, run_sql
+from data_refresh import queries
+
+
+logger = logging.getLogger(__name__)
+
+
+class TableIndex(NamedTuple):
+    index_name: str
+    temp_index_name: str
+    index_def: str
+
+
+def _transform_index_def(
+    existing_index_def: str, temp_table_name: str, table_name: str
+) -> str:
+    """
+    Given a CREATE INDEX statement for an index on the existing media table, return a
+    transformed statement that can be run to apply an identical index to the temp table.
+    """
+    is_pk = "(id)" in existing_index_def
+    tokens = existing_index_def.split(" ")
+
+    # Avoid a naming collision with the existing index by prepending
+    # `temp_import` to the index name. Later when the temp table is promoted,
+    # and the current media table is deleted, the indices will be restored
+    # to their original names.
+    # The index name always follows the word 'INDEX' in the def,
+    # e.g. `CREATE [UNIQUE] INDEX {index_name}`
+    index_name_idx = tokens.index("INDEX") + 1
+    old_index_name = tokens[index_name_idx]
+    # For most indices, we just prefix the index name with `temp_import_`. The pk
+    # index is automatically created at an earlier step and has a different
+    # naming convention.
+    temp_index_name = (
+        f"temp_import_{old_index_name}"
+        if not is_pk
+        else f"temp_import_{table_name}_pkey"
+    )
+    tokens[index_name_idx] = temp_index_name
+
+    # Update the table name to the temp table. The table name always follows
+    # the word 'ON' in the def, e.g. `ON public.audio`
+    table_name_idx = tokens.index("ON") + 1
+    schema_name, table_name = tokens[table_name_idx].split(".")
```

Comment:
It's unlikely (maybe impossible?) for this to result in more than two elements, but it might be good to ensure we only ever get two elements:
```suggestion
    schema_name, table_name = tokens[table_name_idx].split(".", maxsplit=1)
```

Reply:
I don't think it's technically impossible, but would require us to have a table name containing a period, in which case we probably have bigger problems 😆 Absolutely no harm in checking, though, you never know!